opt_einsum/contract.py

```diff
@@ -569,8 +567,11 @@ def _core_contract(operands, contraction_list, backend='auto', evaluate_constant
                 left_pos.append(input_left.find(s))
                 right_pos.append(input_right.find(s))
 
+            # Construct the axes tuples in a canonical order
+            axes = tuple(zip(*sorted(zip(left_pos, right_pos))))
```

Comment:
If one of the lists is empty the return is `(, )` rather than `((,), (,))` causing the Travis failures.